src/catchup/CatchupRange.cpp

```diff
@@ -0,0 +1,96 @@
+// Copyright 2017 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "catchup/CatchupRange.h"
+#include "catchup/CatchupConfiguration.h"
+#include "history/HistoryManager.h"
+#include <cassert>
+#include <utility>
+
+namespace stellar
+{
+
+CatchupRange::CatchupRange(ApplyLedgerChainRange applyLedgerChainRange,
+                           bool applyBuckets)
+    : mApplyLedgerChainRange{std::move(applyLedgerChainRange)}
+    , mApplyBuckets{applyBuckets}
+{
+}
+
+bool
+operator==(CatchupRange const& x, CatchupRange const& y)
+{
+    if (x.mApplyLedgerChainRange != y.mApplyLedgerChainRange)
+    {
+        return false;
+    }
+    if (x.mApplyBuckets != y.mApplyBuckets)
+    {
+        return false;
+    }
+    return true;
+}
+
+bool
+operator!=(CatchupRange const& x, CatchupRange const& y)
+{
+    return !(x == y);
+}
+
+CatchupRange
+makeCatchupRange(CatchupConfiguration const& configuration,
+                 HistoryManager const& historyManager)
+{
+    assert(configuration.toLedger() != CatchupConfiguration::CURRENT);
+
+    auto checkpointContainingToLedger =
+        historyManager.nextCheckpointLedger(configuration.toLedger() + 1) - 1;
```

Comment:
I was expecting `checkpointContainingToLedger` to just be `historyManager.nextCheckpointLedger(configuration.toLedger())` per the formula & comment on what `nextCheckpointLedger` does.

Reply:
It is not unfortunately.
if toLedger = 64 then nextCheckpointLedger = 64, but 64 is in checkpoint 127

I'll make new set of methods for such computation so it is more obvious what is happening.

Reply:
Oh, I see I looked at those helper methods in more detail it's because they operate on "current" instead of LCL, but what is published is LCL... so we end up with the weird +1/-1 here. Don't add new methods as they would only be used here from what I can tell.